src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/bulkCopyTest.java

```diff
@@ -0,0 +1,181 @@
+//  and / or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions :
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS 
+//  IN THE SOFTWARE.
+//---------------------------------------------------------------------------------------------------------------------------------
+package com.microsoft.sqlserver.jdbc.bulkCopy;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.Time;
+import java.sql.Timestamp;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCopy;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.DBConnection;
+import com.microsoft.sqlserver.testframework.DBResultSet;
+import com.microsoft.sqlserver.testframework.DBStatement;
+import com.microsoft.sqlserver.testframework.DBTable;
+import com.microsoft.sqlserver.testframework.sqlType.SqlBit;
+import com.microsoft.sqlserver.testframework.sqlType.SqlDateTime;;
+
+@RunWith(JUnitPlatform.class)
+public class bulkCopyTest extends AbstractTest{
+
+	/**
+	 * Test case for create table functionality {@link DBEngine#createTable(DBTable, com.microsoft.sqlserver.jdbc.SQLServerConnection)}
+	 * @throws Exception 
+	 */
+	@Test
+	public void testBulkCopyWithConnection() throws Exception {
+		
+		DBConnection con = new DBConnection(connectionString);
+		DBStatement stmt = con.createStatement();
+		DBTable sourceTable = new DBTable(true);
+		stmt.createTable(sourceTable);
+		stmt.populateTable(sourceTable);
+		
+		DBTable destinationTable = sourceTable.cloneSchema();
+		stmt.createTable(destinationTable);
+		DBResultSet srcResultSet = stmt.executeQuery("SELECT * FROM "+sourceTable.getEscapedTableName()+";");
+		SQLServerBulkCopy bulkCopy =new SQLServerBulkCopy((Connection)con.product());
+        bulkCopy.setDestinationTableName(destinationTable.getEscapedTableName());
+        bulkCopy.writeToServer((ResultSet)srcResultSet.product());
+        bulkCopy.close();
+        srcResultSet.close();
+        validateValues(con, sourceTable, destinationTable);
+        sourceTable.dropTable(stmt);
+        destinationTable.dropTable(stmt);
```

Comment:
Also here, can we do `DBStatement.dropTable(DBTable)` ? to be consistent with actual JDBC code.